src/decoders.ts

```diff
@@ -167,12 +167,15 @@ const dBoolean = decoder<boolean, unknown>({
   defaultParser: identity,
 });
 
+ // moving the dateValidator logic to here
 const date = decoder<Date, unknown>({
-  typeGuard: (val: unknown): val is Date => val instanceof Date,
-  getDefaultErrorMsg: (val) => `Expected Date but got ${serializeType(val)}`,
+  typeGuard: (val: unknown): val is Date => val instanceof Date && val.toDateString() != "Invalid Date",
```

Comment:
The lint is asking to change double equals to triple but the validation works with double equals only. What should be done?

Reply:
Checkout the difference between === and == . You will get your answer

Reply:
I know that === checks the type and the value whereas == only checks the value.